File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/models/search/searchresult/SearchResultEntityMapper.java
package com.hgianastasio.biblioulbrav2.data.models.search.searchresult;

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper;
import com.hgianastasio.biblioulbrav2.core.search.models.SearchBook;
import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult;
import com.hgianastasio.biblioulbrav2.data.models.search.searchbook.SearchBookEntity;
import com.hgianastasio.biblioulbrav2.data.models.search.searchbook.SearchBookEntityMapper;

import javax.inject.Inject;

/**
 * Created by heitor_12 on 09/05/17.
 */

public class SearchResultEntityMapper extends Mapper<SearchResult,SearchResultEntity> {
    SearchBookEntityMapper entityMapper;

    @Inject
    public SearchResultEntityMapper(SearchBookEntityMapper mapper) {
        this.entityMapper = mapper;
    }

    @Override
    public SearchResultEntity transform(SearchResult input) {
        return null;
    }

    @Override
    public SearchResult transformBack(SearchResultEntity input) {
        SearchResult output = new SearchResult();
        if(input.getScalar()==null){
            output.setResult(entityMapper.tansformBack(input.getResult()));
            output.setCookie(input.getCookie());
            output.setItens(input.getItens());
            output.setPages(input.getPages());
            output.setCurrentPage(input.getCurrentPage());
        }
        output.setEmptyResult(input.getScalar()!=null);
        return output;
    }
}

File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/models/search/searchresult/SearchResultEntityMapper.kt
package com.hgianastasio.biblioulbrav2.data.models.search.searchresult

import com.hgianastasio.biblioulbrav2.core.base.boundaries.Mapper
import com.hgianastasio.biblioulbrav2.core.search.models.SearchResult
import com.hgianastasio.biblioulbrav2.data.models.search.searchbook.SearchBookEntityMapper
import javax.inject.Inject

/**
 * Created by heitor_12 on 09/05/17.
 */
class SearchResultEntityMapper @Inject constructor(var entityMapper: SearchBookEntityMapper) : Mapper<SearchResult, SearchResultEntity>() {
    override fun transform(input: SearchResult): SearchResultEntity? {
        return null
    }

    override fun transformBack(input: SearchResultEntity): SearchResult {
        val output = SearchResult()
        if (input.scalar == null) {
            output.result = entityMapper.tansformBack(input.result!!)
            output.cookie = input.cookie
            output.itens = input.itens
            output.pages = input.pages
            output.currentPage = input.currentPage
        }
        output.isEmptyResult = input.scalar != null
        return output
    }

}